README.md

```diff
@@ -6,18 +6,21 @@
 
 * `NS`: `<clustername>` in DNS zone `<baseDomain>`
 * `A`: `api` in DNS zone `<clustername>` in `AzureCluster` resource group. This record points to the kubernetes API IP of a `cluster`
+* `A`: `bastion` in DNS zone `<clustername>` in `AzureCluster` resource group. This record points the the bastion host IP (managed via a `machineDeployment`) of a `cluster`
 
-> the TTL is set to 1h. This might change in the future or will became configurable.
+> the TTL is set to 5m for the A records and 1h for the NS record. This might change in the future or will became configurable.
 
 ## reconciliation loop
 
 This is a simplified version of the reconciliation loop:
 
 ![](dns-operator.png)
 
-## finalizer
+## finalizer & annotations
 
-The operator set the finalizer `dns-operator-azure.finalizers.giantswarm.io` on the `AzureCluster` CR.
+* The operator set the finalizer `dns-operator-azure.giantswarm.io/azurecluster` on the `AzureCluster` CR.
+* The operator set the finalizer `dns-operator-azure.giantswarm.io/azuremachine` on the `AzureMachine` CR.
```

Comment:
do we really need to handle the deletion of the machine with a finalizer ? can't we just reconcile the dns record if the bastion machine recycle ?

Reply:
We should make sure the finalizer works in both scenario when

*  a Bastion is replaced by adding a new machine before removing the old one
* The old machine is forcibly deleted before the new one is in place

Reply:
> do we really need to handle the deletion of the machine with a finalizer ? can't we just reconcile the dns record if the bastion machine recycle ?

i would like to ensure, that we remove the ip of the deleted bastion machine from the `azureCluster` CR instead of a force-overwrite whenever we thing a valid bastion-machine exist.

Reply:
> With "force-deletion" you mean, delete the machine and also delete the finalizer manually?
just the machine ( say something is wrong with it and i force delete it )